src/Microsoft.ML.OnnxConverter/SaveOnnxCommand.cs

```diff
@@ -276,6 +277,16 @@ private void Run(IChannel ch)
                     Host.Assert(scorePipe.Source == end);
                     end = scorePipe;
                     transforms.AddLast(scoreOnnx);
+
+                    if(rawPred.PredictionKind == PredictionKind.BinaryClassification || rawPred.PredictionKind == PredictionKind.MulticlassClassification)
+                    {
+                        if(scorePipe.Schema.GetColumnOrNull("PredictedLabel")?.Type is KeyDataViewType)
```

Comment:
>is KeyDataViewType [](start = 84, length = 18)

This still doesn't mean that the `KeyToValueMappingTransformer` will work - there can be key type columns that don't have key value annotations. I think there is a utility method called something like `HasKeyNames` that you can use.

Reply:
The `HasKeyValues `method requires me to specify the `keyValueItemType`. Does anyone know if there's a way to avoid having to specify it, since in here I don't care about the keyValueItemType, as long as it has key value annotations.? @ganik  @harishsk (if I leave it null, it defaults to be TextDataViewType)

https://github.com/dotnet/machinelearning/blob/b34c3b6ea5b0d7434dde63012c4c123a85d29dcb/src/Microsoft.ML.Data/Data/SchemaAnnotationsExtensions.cs#L48-L63

Reply:
How about you do something like:

ReadOnlyMemory<char> tmp = default;
if (input.Data.Schema.TryGetAnnotation(TextDataViewType.Instance, AnnotationUtils.Kinds.ScoreValueKind, i,  ref tmp)  && ReadOnlyMemoryUtils.EqualsStr(AnnotationUtils.Const.ScoreValueKind.PredictedLabel, tmp))
                        {

---
In reply to: [382339571](https://github.com/dotnet/machinelearning/pull/4841#discussion_r382339571) [](ancestors = 382339571)

Reply:
More general advice: search for "ScoreValueKind.PredictedLabel" and see how in other places we check for such column

---
In reply to: [382745605](https://github.com/dotnet/machinelearning/pull/4841#discussion_r382745605) [](ancestors = 382745605,382339571)

Reply:
You can look at the implementation of HasKeyValues and filter out the checks that are relying on keyValueItemType . e.g.
```
            if (!(column.Type.GetItemType() is KeyDataViewType keyType))
                return false;
            var metaColumn = column.Annotations.Schema.GetColumnOrNull(AnnotationUtils.Kinds.KeyValues);
            return
                metaColumn != null
                && metaColumn.Value.Type is VectorDataViewType vectorType
                && keyType.Count == (ulong)vectorType.Size)

```

---
In reply to: [382339571](https://github.com/dotnet/machinelearning/pull/4841#discussion_r382339571) [](ancestors = 382339571)

Reply:
I've decided to go with @harishsk suggestion. Please, @yaeldekel let me know if you know about another way of doing this.

---
In reply to: [382804341](https://github.com/dotnet/machinelearning/pull/4841#discussion_r382804341) [](ancestors = 382804341,382339571)